File: interFunk/sort.py

def qsortR(lista):
    if len(lista) == 0:
        return []
    else:
        piv = lista.pop()
        smaller = [a for a in lista if a<=piv]
        bigger  = [b for b in lista if b> piv]
        return qsortR(smaller) + [piv] + qsortR(bigger)

File: interFunk/sort.hs
module Sorting where

qsort :: Ord a => [a] -> [a]
qsort []        = []
qsort (x:xs)    = qsort smaller ++ [x] ++ qsort bigger
        where smaller   = [a | a <- xs, a <= x]
              bigger    = [b | b <- xs, b >  x]

isSorted :: Ord a => [a] -> Bool
isSorted []         = True
isSorted [_]        = True
isSorted (x:xs)
    | x > head xs   = False
    | otherwise     = isSorted xs
